### ISSUE TITLE
topic composition file looks very strange

### ISSUE THREAD

**Issue by vhulden** [mimno/Mallet#41]
This may not be the right place for this question, but the current version of mallet does not produce the #doc name topic proportion configuration of the topic composition. Instead it gives something like this:

`0	file:/(path)/ocr_10.2307_1840442.txt	0.01234185	0.030153958	0.434804596	0.012262633	7.68E-05`

I realize that if this was an actual bug you probably would have noticed it, and thus it is more likely to have something to do with my system, but I don't know where else to ask. Someone's got a question on this on stackoverflow but no answers. Thanks.

**Comment by jarmoza** [comment]
Seconded. Downloaded the latest Mallet yesterday.  I hade to make a slight alteration to some of my code that reads the doc topics file. My presumption is that the topics remain in order from 0...n, but the topic ID (#) is no longer output.

**Comment by vhulden** [comment]
Thanks for the second -- however, I'm not sure I get what you mean that the topics are in order 0...n.  They didn't used to be in order of topic numbers but in order of weight in the document. What mallet used to produce was "topic proportion topic proportion", like this:

`0	file:/path/ocr_10.2307_1840442.txt	16	0.4329549238	2	0.3217675612	18	0.0649332732`

So now I don't know what the numbers are supposed to represent in the new version - what topics do they refer to? I'm using the old version (2.0.7) for the moment.

**Comment by jarmoza** [comment]
Right. So I had forgotten that they were in sorted order like that. My assumption was that with the removal of the topic ID (and as you've pointed out the sorting order), that the order was now just by topic ID. So: [topic 0 weight, topic 1 weight, topic 2 weight....]

**Comment by Eupraxis** [comment]
As it stands that output is useless...if it was a conscious change then I think it should he changed back. My version gives "topic" "weight" which is helpful. 

I can look into the source code to see what is up...which version are we referring to again?

**Comment by vhulden** [comment]
2.0.8 (downloaded from mallet website) and 2.0.9 (latest build cloned from github yesterday). Thanks!

**Comment by dhawaljoh** [comment]
Sorry for bumping this, but, does anyone have an answer yet?

Are they just the probabilities of the topics in order? Or?

I just ran MALLET on 800k documents. (12 Hours). I desperately do not want to re-run the training.

**Comment by christofs** [comment]
Just ran the latest version from Github and see the same behavior. I actually welcome this change because the earlier format was more complex to process. Now it is much more like a regular, sortable table or a data frame. Assuming, of course, the weights are indeed saved in ascending order of the topics. It would have been nice if this change had been documented or explained somehow, somewhere. Or maybe it has?

**Comment by mihaiiancu** [comment]
Add the option --doc-topics-threshold with a value larger than 0.0 and you will get the 'old style' output with the topic ids and percentages  in decreasing order.
This output is OK for visualizing the topics and their distribution across text sources, but I agree with @christofs that the new format is better for processing using a development tool/language.
The issue should be closed since there is no reason to go back.

**Comment by vhulden** [comment]
I did eventually figure it out and agree the new format is better (way
easier for visualization over time, for one). But I would really wish that
a little note could be added somewhere on the documentation page, as most
tutorials were written when the old style format was in effect so people
get incorrect information and this is just plain confusing. A lot of
non-techy people experiment with MALLET...

**Comment by jwr** [comment]
I was just bitten by this and lost hours trying to debug my various processing scripts before finding out that the file format has changed. Turns out the last time I used mallet the format was different.

At the moment I'm not even diving into what the new format is or whether it is sufficient, no time right now. I'll just add the `--doc-topics-threshold` option.

Please — such breaking changes should be documented. The "notes" make no mention of this change (this is where I looked first), neither does the webpage.

**Comment by mimno** [comment]
I'm sorry to hear that -- I was very reluctant to make the change to dense output for exactly this reason. I agree that it's not documented well enough.

**Comment by vhulden** [comment]
Yes, but the change really is a boon - especially in teaching topic modeling to novices, it makes it so easy and neat to draw graphs of a topic over time (well, over an ordered set of documents) with basic spreadsheet software. If someone could just write a paragraph - easily visible to anyone downloading the new version - explaining the change clearly, I think everyone would be happy.

**Comment by jwr** [comment]
I re-read the discussion and I'm not sure what the new format actually is. Is it topic weights ordered by topic, using a dense representation?

**Comment by vhulden** [comment]
Yeah, in the composition file it's the topics in the same order as they appear in the topics file. So if you look at the topics file you have:

```
0 word word word bla bla bla
1 word bla word bla
2 word bla word bla
etc.
```

In the topic composition file, it's just the same order, only horizontally, so you'll have

`file topic0weight topic1weight topic2weight ...`

I usually just manually add a row on top of the composition file with the topic numbers from 0 to however many I'd decided to have, makes it a bit easier to keep track.

**Comment by jwr** [comment]
Ok, thanks — in that case, it doesn't affect me much, I'll just have to rewrite the processing code. The change could indeed be for the better (I don't care either way, but others do), so I'd just like to ask that in the future such format changes be documented, at least in the release notes. This will help people avoid wasting time hunting non-existing bugs.

**Comment by TobyBi** [comment]
Is anyone getting an issue where the topic composition file does not match the topic keys file?
